pkg/azuremachine/validate_create.go

```diff
@@ -32,17 +32,12 @@ func (h *WebhookHandler) OnCreateValidate(ctx context.Context, object interface{
 		return microerror.Mask(err)
 	}
 
-	err = validateLocation(*cr, h.location)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
 	vmcaps, err := h.vmcapsFactory.GetClient(ctx, h.ctrlClient, cr.ObjectMeta)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	supportedAZs, err := vmcaps.SupportedAZs(ctx, cr.Spec.Location, cr.Spec.VMSize)
+	supportedAZs, err := vmcaps.SupportedAZs(ctx, h.location, cr.Spec.VMSize)
```

Comment:
In order to check if the availability zones specified are valid, we use the global location field (ultimately is a flag of the admission controller). This is the only location that we accept for the cluster so it should be ok.

Reply:
Good solution